compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1641,6 +1643,18 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
         NoType
     }
 
+  private[this] def andTypeGen(tp1: Type, tp2: Type, op: (Type, Type) => Type
+    , original: (Type, Type) => Type, isErased: Boolean = ctx.erasedTypes): Type = trace(s"glb(${tp1.show}, ${tp2.show})", subtyping, show = true) {
```

Comment:
it seems `original` is always `_ & _`. Why not inline it?

Reply:
I didn't inline it for consistency reasons: if there is a degree of freedom in the `op` parameter to `liftIfHK`, it would IMO be weird if there wouldn't be a degree of freedom in the `original` as well.

Reply:
On the second thought, I agree, `_ & _` repeated twice doesn't look nice. I have found a compromise in terms of making the `original` in `andTypeGen` a default parameter.

Reply:
I don't understand why this function signature should be consistent with `liftIfHK`. And I don't think having a default parameter is the right way to remove the duplication.

Bonus point: if you inline `original`, you will avoid allocating a lambda most of the time.

Reply:
> I don't understand why this function signature should be consistent with `liftIfHK`.

The problem of the control over the execution flow of a program is solved by decoupling the logic from the data. Doing so partially is inconsistent (even if sufficient for the issue in question).

>Bonus point: if you inline original, you will avoid allocating a lambda most of the time.

How?

Reply:
When calling `andTypeGen`, a lambda is allocated for `original`. But unless `liftIfHK` is called, there is no need to allocate this lambda.

Reply:
Does it have to do with the fact that `andTypeGen` passes the reference along but `liftIfHK` calls it in its body?